agent/hcp/manager.go

```diff
@@ -71,15 +77,80 @@ func NewManager(cfg ManagerConfig) *Manager {
 	}
 }
 
+// runReporter initializes, and if successful, then runs the metrics reporter.
+func (m *Manager) runReporter(ctx context.Context) error {
+	// Step 1: Obtain CCM telemetry configuration
+	// Only enable HCP metrics reporting if server is registered with management plane.
+	telemetryCfg, err := m.cfg.Client.FetchTelemetryConfig(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to obtain CCM telemetry config: %v", err)
+	}
+
+	if telemetryCfg == nil {
+		// Not an error, as the server is not registered with CCM, but early return.
+		return nil
+	}
+
+	// Step 2: Init telemetry.MetricsExporter which sends metrics to HCP Metrics Gateway in OTLP format.
+	// It uses a an OTLP exporter client wrapped by the HCP client.
+	// This enables us to perform HCP auth and easily mock the client interface for tests.
+	// It must first be initialized within the HCP Client with the configured endpoint.
+	if err := m.cfg.Client.InitMetricsClient(ctx, telemetryCfg.Endpoint); err != nil {
+		return fmt.Errorf("failed to init metrics HCP client: %v", err)
+	}
+
+	expCfg := &telemetry.MetricsExporterConfig{
+		Labels: map[string]string{
```

Comment:
Is there a future PR where we get Labels out of `telemetryCfg`? I ask because I think we'll want to tag all metrics with `hcp_internal_id` (possibly organization/project ID as well) so the frontend dashboards can delineate between multiple clusters' metrics in an organization's workspace

Reply:
Related to metric tagging, Cortex/AMP has a feature called "high-availability" that deduplicates metrics given a couple of metric tags: https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-ingest-dedupe.html#Send-high-availability-data. The benefit of this is that if Consul servers and our OTEL collector are both pushing the same metrics, AMP will deduplicate them (saving us $ and avoiding inaccurate metrics).

I think we may want to use something like the below to take advantage of that:

```
"cluster": telemetryCfg.GetLabels()["internal_id"],
"__replica": m.cfg.NodeID,
```

`telemetryCfg.GetLabels()["internal_id"]` is just an example, we could actually set `cluster` in `telemetryCfg.Labels` (the server response)

cc @clly

Reply:
I'll update this in a follow up PR, just to keep the PRs small, with the suggestions made for Cortex, ticket CC-4742! Thanks for catching this and the context with pairing.